src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            addFirst(value);
+        } else {
+            Node<T> newNode = new Node<>(last, value, null);
+            last.next = newNode;
+            last = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        //add first
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == size) { //add last
+            addLast(value);
+        } else { //add middle
+            checkAddIndex(index);
+            Node<T> oldNode = getNodeByIndex(index);
+            Node<T> oldNodePrev = oldNode.prev;
+            Node<T> newNode = new Node<>(oldNodePrev, value, oldNode);
+            oldNode.prev = newNode;
+            oldNodePrev.next = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] arrayList = list.toArray();
+        int length = arrayList.length;
+        if (length == 0) {
```

Comment:
do not create redundant variable
```suggestion
        if (arrayList.length == 0) {
```

Reply:
Yes, I understand it. But, I have done as in the LinkedList.java class (please take a look on photo below). I think they created a redundant variable for better clarity and to make the "if condition" more shorter ...

Reply:
![image](https://user-images.githubusercontent.com/123567653/231527801-6ad69eae-c30d-42de-a93e-1f772a002d26.png)

Reply:
In this method variable `numNew` used 2 times in LinkedList.java class